source/modulo_core/modulo_core/translators/message_writers.py

```diff
@@ -126,6 +131,20 @@ def write_message(message: MsgT, state: StateT):
             message.position = state.get_positions().tolist()
             message.velocity = state.get_velocities().tolist()
             message.effort = state.get_torques().tolist()
+        elif isinstance(message, trajectory.JointTrajectory) and isinstance(state, sr.JointTrajectory):
+            message.joint_names = state.get_joint_names()
+            point = trajectory.JointTrajectoryPoint()
+            for i, point in enumerate(state.get_points()):
+                ros_point = trajectory.JointTrajectoryPoint()
+                ros_point.positions = point.get_positions().tolist()
+                ros_point.velocities = point.get_velocities().tolist()
+                ros_point.accelerations = point.get_accelerations().tolist()
+                ros_point.effort = point.get_torques().tolist()
+                ros_point.time_from_start = rclpy.time.Duration(
+                    seconds=state.get_time_from_start(i).total_seconds()).to_msg()
+                message.points.append(ros_point)
+                message.header.frame_id = state.get_name()
+                # todo: should we properly stamp it?
```

Comment:
note to self and others: we need to discuss this

Reply:
We could but then you'd need to move this code to the `write_stamped_message`